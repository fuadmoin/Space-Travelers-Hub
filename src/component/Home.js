import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux/es/exports';
import { fetchRockets, bookRocket } from '../redux/rocket/rocketSlice';
import RocketBooking from './RocketBooking';
import './home.css';

const Home = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rocket.rockets);
  const status = useSelector((state) => state.rocket);
  const handleRocketBooking = (id) => {
    dispatch(bookRocket(id));
  };
  useEffect(() => {
    if (!rockets.length) dispatch(fetchRockets());
  }, [dispatch]);
  return (
    <div>
      {

      !status.loading && rockets.length ? (

        rockets.map((rocket) => (
          <div key={rocket.id} className="rocketContainer">
            <img className="rocketImage" src={rocket.flickr_images} alt="rocket" />
            <div className="detailContainer" key={rocket.rocket_name}>
              <h1>{rocket.rocket_name}</h1>
              <p>{rocket.description}</p>
              <RocketBooking
                bookRocket={handleRocketBooking}
                id={rocket.id}
                status={rocket.reserved}
              />
            </div>
          </div>
        ))) : null

    }

    </div>
  );
};

export default Home;
