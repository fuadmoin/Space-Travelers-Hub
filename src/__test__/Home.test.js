import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Home from '../component/Home';

const mockStore = configureMockStore();
const store = mockStore({
  rocket: {
    rockets: [
      {
        id: '1',
        name: 'Falcon 1',
        description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009',
        flickr_images: 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
        reserved: false,
      },
      {
        id: '2',
        name: 'Falcon 9',
        description: 'Falcon 9 is a partially reusable two-stage-to-orbit medium lift launch vehicle designed and manufactured by SpaceX.',
        flickr_images: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
        reserved: true,
      },
    ],
    isLoading: false,
    error: null,
  },
});

describe('Home', () => {
  it('should render the Home component', () => {
    const tree = renderer.create(<Provider store={store}><Home /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render rocket information', () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const rocketNameElement = screen.getAllByText(/Falcon 1/i);
    const rocketDescriptionElement = screen.getByText(/The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009/i);
    const rocketImageElement = screen.getAllByAltText('rocket');

    expect(rocketNameElement[0]).toBeInTheDocument();
    expect(rocketDescriptionElement).toBeInTheDocument();
    expect(rocketImageElement[0]).toBeInTheDocument();
  });
});
