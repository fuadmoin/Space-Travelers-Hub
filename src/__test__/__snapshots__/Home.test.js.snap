// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the Home component 1`] = `
<div>
  <div
    className="rocketContainer"
  >
    <img
      alt="rocket"
      className="rocketImage"
      src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
    />
    <div
      className="detailContainer"
    >
      <h1>
        Falcon 1
      </h1>
      <p>
        The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009
      </p>
      <button
        className="btnReserve"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
  <div
    className="rocketContainer"
  >
    <img
      alt="rocket"
      className="rocketImage"
      src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
    />
    <div
      className="detailContainer"
    >
      <h1>
        Falcon 9
      </h1>
      <p>
        Falcon 9 is a partially reusable two-stage-to-orbit medium lift launch vehicle designed and manufactured by SpaceX.
      </p>
      <button
        className="btnReserve"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderBottom": "1px solid black",
            "borderLeft": "1px solid grey",
            "borderRight": "1px solid black",
            "borderTop": "1px solid grey",
            "color": "grey",
            "width": "150px",
          }
        }
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;
